### PR TITLE
Forward Backward Difference explained with examples.

### PR DESCRIPTION
**Pull request by rajflume**
Forward backward difference using TFF. Explained with examples.

### CODE REVIEW

File: tf_quant_finance/examples/jupyter_notebooks/Forward_Backward_Diff.ipynb [hunk 1]
```diff
@@ -0,0 +1,247 @@
+{
+ "cells": [
```
Comment (cyrilchim) [inline tf_quant_finance/examples/jupyter_notebooks/Forward_Backward_Diff.ipynb]: Could you please add licence and upgrade-install statements (Upgrade to TF 2.1+ , Install TFF, QuantLib), like [here](https://colab.sandbox.google.com/github/google/tf-quant-finance/blob/master/tf_quant_finance/examples/jupyter_notebooks/American_Option_Black_Scholes.ipynb)

File: tf_quant_finance/examples/jupyter_notebooks/Forward_Backward_Diff.ipynb [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import tensorflow as tf\n",
+    "import functools\n",
+    "import tf_quant_finance.math.gradient as tff_gradient\n",
+    "import tf_quant_finance.math.optimizer as tff_optimizer\n",
+    "import tensorflow_probability as tfp"
```
Comment (cyrilchim) [inline tf_quant_finance/examples/jupyter_notebooks/Forward_Backward_Diff.ipynb]: Please remove 
import tensorflow_probability as tfp

File: tf_quant_finance/examples/jupyter_notebooks/Forward_Backward_Diff.ipynb [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
...2lMnHmPW0gwefOvHJM+OYNvLB/jov8uRO5VIJBKS82N4avMvSCmI4bm1m3j5dx9gtzvw8Vez/tXbmb0ik9ee3Mqrv9+C0+kkvTCW9X/7MeMGMw/f9BL9nSP4h2h45JUf4any4NGfvEb38SGiEoJ44E/XM9ir4/f/7y0mrDau/9kiii5J5rU/fUbFF/XYpDqWrU6l9dgQtSUjpGUmcM9vV9Lboaf5kJkFlxSy+vZiGo708sZfP8UmHSCzIJj33iyloqSW1bcU4unlzgtPf0pQsIYfXlfIvj0NHKnu4Ee3zcXpcPLGaweYNz+R+IRA3nyjBH9/NUsWp/LJjqMYDRa+f2UOFQePc7x9iKUL46mt66VvwMQPVuZztL6PvoExli1M4YuyZpQKOXnpEXyyp45ZWZFo1Aq2761n6ZwkrBN29le18b2CeCpqO4gJ9UWrVrK3qpW52TGMjJkY0Y8TF+pHSW07hSkRGM1WxsYtBGlVNPTqmZ0eR4/OxJK5BQD0GG2EaRTUHO8lQqOkf8yEY1wHQN/IKAq5jJaBEe5dUszyjIQzuq7OBk6nk97eXioqKvDz8yM7O/ucjcnd3d3x9/cnJiaGzMxMioqKyM7OJjAwEJvNxvHjxykrK6O0tJTa2lo6OjoYHR0VAySLxUJ1dbWYUnCZzriiYblcjru7OwqFAnd3d9zd3XE4HDz11FN0dHTg4eGBu7v7jL1Murq6pimZwsPD6erqOu0+MplMTHmcT1w0pHuqJ5FrtlJTUxN+fn7MmjVrRubIp8KZkq5Op6OiooK+vj5ycnLE4XlRUVH4+/uj0+mora3lwIEDdA+288MHvse9L94IwO+ue45n73+Tvu5+Dh48yLjFwIMb7uTeZ26ivb6H+1c8RfXeegC8fb148JXbuPxH89jyt7384c6NWExW3D3k3PvH61h+02w+3LCXl9a/j8PhID4jnN++8mPM41bW/2gDI/16AkJ8+O1LN4PTyWN3vIpu2EhyTiR3PfZ96qo6ePE/tyCRwE8fvoLAcA3/u+49pE4Pbr1rJQsvy+D/NnzBsaoOZs1LYNlVOWx+q4ym2j5WrZ5NblEsuzY3ExeTyj1rr8RL5cEbL5XQ29fBnEVhHKvp4v/+vpvi+eH4+at45cU9BAZ6s+KKLHZ9dozurhFuvLGY/v4xdn52jGt+mI/TCe9/cIgVl2agVMjZ8Led
```
Comment (cyrilchim) [inline tf_quant_finance/examples/jupyter_notebooks/Forward_Backward_Diff.ipynb]: Maybe remove as this one is not the vector-function you are differentiating?

Comment (cyrilchim) [inline tf_quant_finance/examples/jupyter_notebooks/Forward_Backward_Diff.ipynb]: Could you please also add a comment that you can pass the coefficients [w_1, w_2] (output_gradient for tff.math.gradient and input_gradient for tff.math.fwd_gradients)?

Comment (cyrilchim) [inline tf_quant_finance/examples/jupyter_notebooks/Forward_Backward_Diff.ipynb]: + u_3 y

Comment (cyrilchim) [inline tf_quant_finance/examples/jupyter_notebooks/Forward_Backward_Diff.ipynb]: + u_3 y

File: tf_quant_finance/examples/jupyter_notebooks/Forward_Backward_Diff.ipynb (hunk 1 above)
Comment (rajflume) [reply tf_quant_finance/examples/jupyter_notebooks/Forward_Backward_Diff.ipynb]: added

File: tf_quant_finance/examples/jupyter_notebooks/Forward_Backward_Diff.ipynb (hunk 3 above)
Comment (rajflume) [reply tf_quant_finance/examples/jupyter_notebooks/Forward_Backward_Diff.ipynb]: removed image

Comment (rajflume) [reply tf_quant_finance/examples/jupyter_notebooks/Forward_Backward_Diff.ipynb]: corrected

Comment (rajflume) [reply tf_quant_finance/examples/jupyter_notebooks/Forward_Backward_Diff.ipynb]: corrected

Comment (rajflume) [reply tf_quant_finance/examples/jupyter_notebooks/Forward_Backward_Diff.ipynb]: Added comments at the end

File: tf_quant_finance/examples/jupyter_notebooks/Forward_Backward_Diff.ipynb (hunk 2 above)
Comment (rajflume) [reply tf_quant_finance/examples/jupyter_notebooks/Forward_Backward_Diff.ipynb]: removed

**Pull request merged**